bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2HttpClient.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enigma2.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * The {@link Enigma2HttpClient} class is responsible for sending HTTP-Get requests to the Enigma2 device.
+ * It is devided from {@link Enigma2Client} for better testing purpose.
+ *
+ * @author Guido Dolfen - Initial contribution
+ */
+@NonNullByDefault
+public class Enigma2HttpClient {
+    private int timeout;
+
+    public Enigma2HttpClient(int timeout) {
+        this.timeout = timeout;
+    }
+
+    public String get(String url) throws IOException, IllegalArgumentException {
+        String xml = HttpUtil.executeUrl( "GET", url, timeout * 1000);
+        // remove some unsupported xml-characters
+        Pattern p = Pattern.compile("[^\\u0009\\u000A\\u000D\\u0020-\\uD7FF\\uE000-\\uFFFD\\u10000-\\u10FFF]+");
+        return p.matcher(xml).replaceAll("");
+    }
```

Comment:
I see why this needs to be its own class. You could replace this class with a static method `get(String url, int timeout)`.

Reply:
If it ist a static method, I could not mock it with Mockito. In case of that, I'm not able to test the code because I should not send a real HTTP request in test cases. Any suggestions?